### PR TITLE
Implement pre-commits

### PR DESCRIPTION
**Pull request by ViacheslavDanilov**

### CODE REVIEW

**Comment by ViacheslavDanilov** [comment]
@ryndovaira @mak-en @trioxane 

I implemented pre-commits allowing us to use the same standards during coding. Please have a look at the pre-commit pipeline described at: 

1. https://pre-commit.com/index.html
2. https://towardsdatascience.com/4-pre-commit-plugins-to-automate-code-reviewing-and-formatting-in-python-c80c6d2e9f5

**Review by ryndovaira — DISMISSED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  | buck-out
+  | build
+  | dist
+)/
+'''
+
+[tool.interrogate]
+fail-under = 50
+ignore-init-method = true
+ignore-init-module = false
```
Comment (mak-en) [inline pyproject.toml:22]: true

**Review by mak-en — CHANGES_REQUESTED** [review]

**Comment by ViacheslavDanilov** [comment]
@mak-en 

Please check the replacement of **interrogate** with **pydocstyle**

**Comment by ViacheslavDanilov** [comment]
@ryndovaira @mak-en 

No worries about merge conflicts. I will resolve them by myself.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  | \.venv
+  | _build
+  | buck-out
+  | build
+  | dist
+)/
+'''
+
+[tool.interrogate]
+fail-under = 50
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ignore-init-method = true
+ignore-init-module = false
+ignore-magic = false
+ignore-semiprivate = false
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ignore-private = false
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ignore-property-decorators = false
+ignore-module = true
+ignore-nested-functions = false
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ignore-nested-classes = true
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ignore-setters = false
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: .pre-commit-config.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  hooks:
+    - id: black
+
+# flake8
+-   repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+    -   id: flake8
+
+# pylint
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply .pre-commit-config.yaml]: Fixed

File: pyproject.toml (hunk 1 above)
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  | build
+  | dist
+)/
+'''
+
+[tool.interrogate]
+fail-under = 50
+ignore-init-method = true
+ignore-init-module = false
+ignore-magic = false
```
[the remark this answers is not in the document]
Comment (ViacheslavDanilov) [reply pyproject.toml]: Fixed

**Comment by ViacheslavDanilov** [comment]
> @ryndovaira @mak-en
> 
> No worries about merge conflicts. I will resolve them by myself.

Resolved

**Review by mak-en — COMMENTED** [review]
Looks good!

**Comment by mak-en** [comment]
> 

I am on that

**Review by mak-en — DISMISSED** [review]

**Pull request merged**